app/src/main/java/ar/com/wolox/android/cookbook/scanqr/ScanQrActivity.kt

```diff
@@ -0,0 +1,68 @@
+package ar.com.wolox.android.cookbook.scanqr
+
+import android.app.Activity
+import android.content.Intent
+import android.util.Log
+import ar.com.wolox.android.cookbook.R
+import ar.com.wolox.android.cookbook.scanqr.captureqr.CaptureQrActivity
+import ar.com.wolox.android.cookbook.scanqr.scanerror.ScanErrorFragment
+import ar.com.wolox.android.cookbook.scanqr.scanmenu.ScanMenuFragment
+import ar.com.wolox.android.cookbook.scanqr.scansuccess.ScanSuccessFragment
+import ar.com.wolox.wolmo.core.activity.WolmoActivity
+import com.google.zxing.integration.android.IntentIntegrator
+import javax.inject.Inject
+
+class ScanQrActivity : WolmoActivity(), ScanQrView {
+
+    @Inject
+    lateinit var presenter: ScanQrPresenter
+
+    override fun layout(): Int = R.layout.activity_scan_qr
+
+    override fun init() {
+    }
+
+    override fun showSuccessScreen(result: String) = replaceFragment(R.id.vFragmentHolder, ScanSuccessFragment.newInstance(result))
+
+    override fun showErrorScreen() = replaceFragment(R.id.vFragmentHolder, ScanErrorFragment())
+
+    override fun showScannerView() {
+        IntentIntegrator(this).run {
+            /*
+                ADD ANY EXTRA THAT YOU WANT TO PASS TO CAPTURE QR ACTIVITY HERE
+             */
+            setDesiredBarcodeFormats(IntentIntegrator.QR_CODE) // --> DEFINE THE FORMAT OF THE CODE YOUR SCANNING
```

Comment:
Try to use this convention to comments and documentation: https://kotlinlang.org/docs/reference/coding-conventions.html#documentation-comments :) In case you considere there not really a good "reason" to keep these commets, then just delete them :)